lib/engine/game/g_18_tn.rb

```diff
@@ -80,10 +80,13 @@ def operating_round(round_num)
       def routes_revenue(routes)
         total_revenue = super
 
-        abilities = routes.first&.corporation&.abilities(:civil_war)
+        corporation = routes.first&.corporation
 
-        return total_revenue if !abilities || abilities.empty?
+        abilities = corporation&.abilities(:civil_war)
 
+        return total_revenue if !abilities || abilities.empty? || routes.size < corporation&.trains&.size
```

Comment:
why don't you do a return unless corporation first

routes.size < corporations&trins&.size will crash if there is no corporation

3 < nil is undefined

Reply:
Ah, I thought &. would short cut the comparision.

Anyway, it could never happen as if corporation is nil, then abilities would be nil from line 85.
So I have removed both &. in that last one.

Or do you prefer to add a
  return total_revenue unless corporation
before line 85, and remove &. from line 85?